ws/api/transport.go

```diff
@@ -0,0 +1,91 @@
+package api
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/go-zoo/bone"
+	"github.com/gorilla/websocket"
+	"github.com/mainflux/mainflux"
+	manager "github.com/mainflux/mainflux/manager/client"
+	"github.com/mainflux/mainflux/ws"
+)
+
+const protocol = "ws"
+
+var (
+	errUnauthorizedAccess = errors.New("missing or invalid credentials provided")
+	upgrader              = websocket.Upgrader{
+		ReadBufferSize:  1024,
+		WriteBufferSize: 1024,
+		CheckOrigin: func(r *http.Request) bool {
+			return true
+		},
+	}
+	auth manager.ManagerClient
+)
+
+// MakeHandler returns http handler with handshake endpoint.
+func MakeHandler(svc ws.Service, mc manager.ManagerClient) http.Handler {
+	auth = mc
+
+	mux := bone.New()
+	mux.GetFunc("/channels/:id/messages", handshake(svc))
+
+	return mux
+}
+
+func handshake(svc ws.Service) func(http.ResponseWriter, *http.Request) {
```

Comment:
It feels like this function is doing too much. As far as I understand the code, it handles both the connection establishing and listening to its "events".

Reply:
@anovakovic01 this remark still holds. Please take care of it! Also, consider using `http.HandlerFunc` as return type.